### PR TITLE
Enable access to the last error value in last_error_context

### PR DESCRIPTION
**Pull request by NatiMiller**
Add an option to get the last error kept in object. (In order to be displayed by log messages.)

### CODE REVIEW

**Comment by sylveon** [comment]
I feel like `get_last_error()` is a misnomer because the actual last error might have changed since the object was created.

**Comment by NatiMiller** [comment]
Hi Charles,
Function get_last_error() is required and is correct.
Sometimes user wants to get last error that was saved by object and was taken following specific action, even though the current last error is different.
The aim of this object is to save last error at specific time, while probably the real last error is different at time of destruction of this object.

**Comment by dunhor** [comment]
I'll agree with @sylveon here. Something like `value()` seems like a better name for the function.

File: include/wil/resource.h [hunk 1]
```diff
@@ -72,6 +72,11 @@ namespace wil
         {
         }
 
+        DWORD get_last_error()
+		{
```
Comment (dunhor) [inline include/wil/resource.h]: Spaces please. And ideally order after the destructor (or at least not sandwiched between two constructors)

Comment (NatiMiller) [reply include/wil/resource.h]: Agree.  Fix spaces , change location at file, and change name of function to get_saved_error.

Comment (NatiMiller) [reply include/wil/resource.h]: Agree.  Fix spaces , change location at file, and change name of function to get_saved_error.

File: include/wil/resource.h [hunk 2]
```diff
@@ -72,6 +72,11 @@ namespace wil
         {
         }
 
+        auto value() const
+		{
```
Comment (ChrisGuzak) [inline include/wil/resource.h:76]: this is indented too much, perhaps using tabs instead of spaces

**Review by ChrisGuzak — CHANGES_REQUESTED** [review]
Update the PR title to describe the functionality, e.g. "Enable access to the last error value in last_error_context"

**Review by ChrisGuzak — APPROVED** [review]

**Pull request merged**